.github/workflows/doc-automation.yml

```diff
@@ -21,6 +21,10 @@ jobs:
         cd ./docs/sphinx
         make html
       id: build
+    - name: Check links in all markdown files
```

Comment:
So the problem with this approach is `doc-automation` runs on merge to master because we can't change our documentation based on whatever commits anyone makes

My suggestion would be to make this work on `lint.yml`. 

Also we need to make sure that the if the output of those scripts is not empty then the action needs to fail (only in lint.yml). Also if you are running the markdown link checker did it print broken links that you were then able to fix? 

Finally for broken links in general I think it's fine to fix them all at once in a single PR but for typos in documentation I don't think those are critical to fix in batch and we should instead just show users typos in the files that they changed only (I do this already to check for linting so just borrow same idea)

Reply:
So you want both spellchecking and link-checking to be in `lint.yml`. As for link-checking, you want the whole documentation to be checked, but for spellchecking just the modified files. Am I correct?
For link-checking, how is it to use `gaurav-nelson/github-action-markdown-link-check` as a step in in the `build` job?

For spellchecking we could add a dedicated job to `lint` workflow for it to run in parallel or do it as a step in the same `build` job.